tactics/class_tactics.ml

```diff
@@ -1424,22 +1425,28 @@ let deps_of_constraints cstrs evm p =
     Intpart.union_set (Evar.Set.union evx evy) p)
     cstrs
 
+(** get dependencies of undefined evars in evar map
+    compute new dependency caches along the way *)
 let evar_dependencies pred evm p =
   Evd.fold_undefined
-    (fun ev evi _ ->
+    (* evm_caches is evar map with updated caches threaded through the fold *)
+    (fun ev evi evm_with_caches ->
```

Comment:
Nitpick: I'd rather call the threaded evarmap just `evm`. That's standard practice in Coq to shadow the name of an old state when it's not relevant anymore.